### PR TITLE
Add Rust examples

### PR DESCRIPTION
**Pull request by nxsaken**
This is a draft PR on new Rust SDK examples.

### CODE REVIEW

File: Rust/configs/bob_chess.toml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+torii_url = "http://127.0.0.1:8080/"
+
+[basic_auth]
+web_login = "mad_hatter"
+password = "ilovetea"
+
+[account]
+domain = "chess"
+public_key = "ed012004FF5B81046DDCCF19E2E451C45DFB6F53759D4EB30FA2EFA807284D1CC33016"
+private_key = "802640AF3F96DEEF44348FEB516C057558972CEC4C75C4DB9C5B3AAC843668854BF82804FF5B81046DDCCF19E2E451C45DFB6F53759D4EB30FA2EFA807284D1CC33016"
```
Comment (0x009922) [inline Rust/configs/bob_chess.toml]:
```suggestion
[account]
domain = "chess"
# Bob's key pair
public_key = "ed012004FF5B81046DDCCF19E2E451C45DFB6F53759D4EB30FA2EFA807284D1CC33016"
private_key = "802640AF3F96DEEF44348FEB516C057558972CEC4C75C4DB9C5B3AAC843668854BF82804FF5B81046DDCCF19E2E451C45DFB6F53759D4EB30FA2EFA807284D1CC33016"
```

File: Rust/examples/account_register.rs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    let register_account = Register::account(Account::new(account.clone()));
+    as_who.submit_blocking(register_account)?;
+    // Observe that the account has really been registered.
+    let account = as_who.request(account::by_id(account))?;
+    println!(
+        "---------------\n\
+        `account_id`: {}\n\
+        Registered by {}",
+        account.id, as_who.account,
+    );
```
Comment (0x009922) [inline Rust/examples/account_register.rs]: - This printing format is convenient for CLI, but is a bit noisy for educational purposes, imo
- Would be useful to see a sample output right here
- In Iroha itself, we recently started to follow a convention of omitting `_id` suffix in most cases

```suggestion
    println!("Account: {}\nRegistered by: {}", account.id, as_who.account);
    // Account: ed12...41@wonderland
    // Registered by: ed01...12@wonderland
```

**Review by 0x009922 — COMMENTED** [review]
New examples look indeed more expressive with a thin higher-level layer. However, I think it is problematic that we will have to introduce readers into these examples-specific abstractions. Although it is now certainly a step towards better examples, I think it is a sign that something like this should be a part of `iroha` library API. In my view, best examples would be "bare", self-contained, and would not require extra layers apart from the target library itself.

**Comment by nxsaken** [comment]
@0x009922 I definitely went overboard with the asset definitions, realized I can remove all that stuff. I guess the pain point that led to this is the lack of const APIs for constructing data primitives.

File: Rust/configs/alice_chess.toml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+chain = "00000000-0000-0000-0000-000000000000"
```
Comment (mversic) [inline Rust/configs/alice_chess.toml:1]: we emphasize somewhere in the docs that this value **must be** generated as UUID (and why). This here is just a dummy value

File: Rust/configs/alice_chess.toml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+torii_url = "http://127.0.0.1:8080/"
+
+[basic_auth]
+web_login = "mad_hatter"
+password = "ilovetea"
+
+[account]
+# Domain in which Alice has an authority
```
Comment (mversic) [inline Rust/configs/alice_chess.toml:9]: what does "has an authority" mean here? I'd rephrase

**Review by mversic — CHANGES_REQUESTED** [review]

File: Rust/examples/account_register.rs [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+fn main() -> iroha_examples::Result<()> {
+    // An account is created for a signatory in a domain.
+    // By default, only the owner of the domain can register accounts in it.
+    let as_alice_in_wland = AliceInWonderland::client();
+    // The same signatory can have an account in different domains.
+    register(&as_alice_in_wland, AliceInChess::id())?;
+
+    // The domain owner can also grant a permission to register accounts in the domain.
+    let can_register_accounts_in_chess = CanRegisterAccountInDomain {
```
Comment (mversic) [inline Rust/examples/account_register.rs]:
```suggestion
    let can_register_accounts_in_chess = CanRegisterAccount {
```

File: Rust/Cargo.toml [hunk 6]
```diff
@@ -14,17 +14,7 @@ keywords = ["blockchain", "crypto", "iroha", "ledger"]
...
-test_network = { "git" = "https://github.com/hyperledger/iroha.git", branch = "iroha2-dev" }
-
-eyre = "0.6.8"
-
-serde = { version = "1.0.151", default-features = false }
-serde_json = { version = "1.0.91", default-features = false }
-
-tokio = "1.23.0"
+iroha = { git = "https://github.com/hyperledger/iroha.git" }
+iroha_executor_data_model = { git = "https://github.com/hyperledger/iroha.git" }
```
Comment (mversic) [inline Rust/Cargo.toml]:
```suggestion
iroha = { version = "=2.0.0-rc.1.0", git = "https://github.com/hyperledger/iroha.git" }
iroha_executor_data_model = { version = "=2.0.0-rc.1.0", git = "https://github.com/hyperledger/iroha.git" }
```

Comment (mversic) [reply Rust/Cargo.toml]: until we publish on `crates.io`

File: Rust/examples/account_register.rs (hunk 5 above)
Comment (mversic) [reply Rust/examples/account_register.rs]: if you run `cargo check` this should fail

File: Rust/examples/account_unregister.rs [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+fn unregister(as_who: &Client, account: AccountId) -> iroha_examples::Result<()> {
+    let unregister_account = Unregister::account(account.clone());
+    as_who.submit_blocking(unregister_account)?;
+    // Observe that the account has really been unregistered.
+    as_who
+        .query(FindAccounts)
+        .filter_with(|acc| acc.id.eq(account.clone()))
+        .execute_single()
+        .expect_err("account should not be found");
+    println!("Account: {}\nUnregistered by: {}", account, as_who.account);
```
Comment (mversic) [inline Rust/examples/account_unregister.rs]:
```suggestion
    println!("Account: {account}\nUnregistered by: {}", as_who.account);
```

File: Rust/examples/asset_definition_register.rs [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+        &as_alice_in_wland,
+        AssetDefinition::store(ChessBook::id())
+            .with_logo("QmQqzMTavQgT4f4T5v6PWBp7XNKtoPmC9jvn12WPT3gkSE".parse::<IpfsPath>()?)
+            .with_metadata(Metadata::default()),
+    )?;
+    Ok(())
+}
+
+fn register(as_who: &Client, asset_definition: NewAssetDefinition) -> iroha_examples::Result<()> {
+    let asset_definition_id = asset_definition.id().clone();
```
Comment (dima74) [inline Rust/examples/asset_definition_register.rs:81]: This currently doesn't compile, getters should be added to `NewAssetDefinition`

https://github.com/hyperledger-iroha/iroha/issues/5237

Comment (dima74) [reply Rust/examples/asset_definition_register.rs:81]: Fixed by using `Identifiable` trait
